lxd/main_init.go

```diff
@@ -1,1266 +1,507 @@
 package main
 
 import (
-	"encoding/pem"
 	"fmt"
-	"net"
-	"os"
-	"os/exec"
-	"path/filepath"
-	"strconv"
-	"strings"
-	"syscall"
 
-	"golang.org/x/crypto/ssh/terminal"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
 
 	"github.com/lxc/lxd/client"
 	"github.com/lxc/lxd/lxd/cluster"
 	"github.com/lxc/lxd/lxd/util"
-	"github.com/pkg/errors"
-
 	"github.com/lxc/lxd/shared"
 	"github.com/lxc/lxd/shared/api"
-	"github.com/lxc/lxd/shared/cmd"
-	"github.com/lxc/lxd/shared/idmap"
-	"github.com/lxc/lxd/shared/logger"
 )
 
-// CmdInit implements the "lxd init" command line.
-type CmdInit struct {
-	Context         *cmd.Context
-	Args            *Args
-	RunningInUserns bool
-	VarDir          string
-	PasswordReader  func(int) ([]byte, error)
+type initData struct {
+	api.ServerPut `yaml:",inline"`
+	Cluster       *initDataCluster       `json:"cluster" yaml:"cluster"`
+	Networks      []api.NetworksPost     `json:"networks" yaml:"networks"`
+	StoragePools  []api.StoragePoolsPost `json:"storage_pools" yaml:"storage_pools"`
+	Profiles      []api.ProfilesPost     `json:"profiles" yaml:"profiles"`
 }
 
-// Run triggers the execution of the init command.
-func (cmd *CmdInit) Run() error {
-	// Check that command line arguments don't conflict with each other
-	err := cmd.validateArgs()
-	if err != nil {
-		return err
-	}
+type initDataCluster struct {
+	api.ClusterPut  `yaml:",inline"`
+	ClusterPassword string `json:"cluster_password" yaml:"cluster_password"`
+}
 
-	// Connect to LXD
-	path := ""
-	if cmd.VarDir != "" {
-		path = filepath.Join(cmd.VarDir, "unix.socket")
-	}
-	client, err := lxd.ConnectLXDUnix(path, nil)
-	if err != nil {
-		return fmt.Errorf("Unable to talk to LXD: %s", err)
-	}
+type cmdInit struct {
+	cmd    *cobra.Command
+	global *cmdGlobal
 
-	existingPools, err := client.GetStoragePoolNames()
-	if err != nil {
-		// We should consider this fatal since this means
-		// something's wrong with the daemon.
-		return err
-	}
+	flagAuto    bool
+	flagPreseed bool
 
-	data := &cmdInitData{}
+	flagNetworkAddress  string
+	flagNetworkPort     int
+	flagStorageBackend  string
+	flagStorageDevice   string
+	flagStorageLoopSize int
+	flagStoragePool     string
+	flagTrustPassword   string
+}
 
-	// Kick off the appropriate way to fill the data (either
-	// preseed, auto or interactive).
-	if cmd.Args.Preseed {
-		err = cmd.fillDataPreseed(data, client)
-	} else {
-		// Copy the data from the current default profile, if it exists.
-		cmd.fillDataWithCurrentServerConfig(data, client)
+func (c *cmdInit) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = "init"
+	cmd.Short = "Configure the LXD daemon"
+	cmd.Long = `Description:
+  Configure the LXD daemon
+`
+	cmd.Example = `  init --preseed
```

Comment:
Spaces after the opening "`" are intentional?

Reply:
Yes